src/features/swap/index.tsx

```diff
@@ -126,8 +133,12 @@ const SwapWidget = ({ sell }: Params) => {
       {
         event: CowEvents.ON_CHANGE_TRADE_PARAMS,
         handler: (newTradeParams) => {
-          const { orderType: tradeType } = newTradeParams
-          dispatch(setSwapParams({ tradeType }))
+          const { orderType: tradeType, recipient } = newTradeParams
+          if (recipient && isBlockedAddress(recipient)) {
+            setBlockedAddress(recipient)
+          } else {
+            dispatch(setSwapParams({ tradeType }))
+          }
```

Comment:
nit: I would move this a bit around. I would always disptach a setSwapParams(this is indipendent of the recipient). Then afterwards check if we need to block the recipient.